qiskit/circuit/random/utils.py

```diff
@@ -146,8 +146,14 @@ def random_circuit(
 
             # with some low probability, condition on classical bit values
             if conditional and rng.choice(range(10)) == 0:
-                value = rng.integers(0, np.power(2, num_qubits))
-                op.condition = (cr, value)
+                parts = rng.integers(0, 1<<16, size=4)
+                shift = 0
+                condition_int = 0
+                for part in parts:
+                    ipart = (int)(part)
+                    condition_int += ipart << shift
+                    shift += 16
+                op.condition = (cr, condition_int)
```

Comment:
Shouldn't this be based on the number of qubits? We want to pick a random integer between 0 and `2**num_qubits` as the random condition state. Right now this is hard coded as a random large integer. Also I think this can return negatives which we don't support for the value.

Reply:
Thanks for the suggestion. I agree that this should be based on the number of qubits, so I did the following:
```python
parts = rng.integers(0, 1<<16, size=int(num_qubits/16))
shift = 0
condition_int = 0
for part in parts:
    ipart = int(part)
    condition_int += ipart << shift
    shift += 16
condition_int = condition_int<<(num_qubits%16)
value_add = rng.integers(0, 1<<(num_qubits%16))
condition_int += int(value_add)
```
Here, for any number of qubits this code generates a random number with maximum value of 2**num_qubits - 1. For example, if num_qubits is 70, then the loop runs 4 times generating a random number upto 2^64 and then that number is shifted bitwise by 6 bits (70-64) and another random number of 6 bits is added to it. Kindly let me know if this approach is correct.

Reply:
Hi @mtreinish, just wanted to remind you about the changes I did after your suggestion. Could you please review the code above so that I can commit the changes to the PR if the code is correct?

Reply:
Sorry for the delay, that approach looks reasonable to me. Please make the changes to your PR branch and probably add some unit tests at the same time to verify that for large circuits we still can set a condition. Then I think with a release not this will be ready to go